chain/bitcoind_conn.go

```diff
@@ -370,6 +415,97 @@ func (c *BitcoindConn) blockEventHandler() {
 	}
 }
 
+// blockEventHandlerRPC is a goroutine that uses the rpc client to check if we
+// have a new block every so often.
+func (c *BitcoindConn) blockEventHandlerRPC() {
+	defer c.wg.Done()
+
+	log.Info("Started polling for new bitcoind blocks via RPC.")
+	ticker := time.NewTicker(c.pollBlockTimer)
+
+	for {
+		select {
+		case <-ticker.C:
+			// Every so many seconds, we poll to see if there's a new
+			// block.
+			blockHash, err := c.client.GetBestBlockHash()
+			if err != nil {
+				log.Errorf("Unable to retrieve best block "+
+					"hash: %v", err)
+			}
```

Comment:
I haven't dug into the function `c.client.GetBestBlockHash`, but how does this handle errors? Like a 500 from the RPC or something? Just looking at this, it seems like we just log an error and continue? I wonder if there's a better way to handle it, like retrying a failed request.

Reply:
Good call! The reason I didn't do a retry was because I figured we'd be trying again in X seconds anyway. Maybe we can retry the request every second for X - 1 seconds?

Reply:
Yeah that's true. What do you think about maybe 1 retry after failure that has a 1 second pause after the failure? Then if that second one fails, maybe we just log an error message but let it continue to the next cycle?

Reply:
Yeah that sounds good to me